### PR TITLE
Test only: Speed up MoE tests

### PR DESCRIPTION
**Pull request by amorrisonTT**
### Summary
Likely due to #45368 , `test_moe_compute_6U.py` had ballooned to over an hour of runtime
- Made sure all of the MoE tests were taking advantage of threads for torch stuff
- Added prepped weight and reference caching between analogous test cases so these slow to generate quantities could be re-used
- Skip, rather than xfail, known failing (slow) test cases
Got this test down to < 30 minutes 

### Notes for reviewers
We will like still need to bump the e2e MoE pipeline a little bit

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:amorrison/speed-up-moe-tests)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:amorrison/speed-up-moe-tests)

### CODE REVIEW

File: tests/nightly/tg/ccl/moe/test_selective_combine_6U.py [hunk 1]
```diff
@@ -12,6 +12,11 @@
 import torch
 import ttnn
 
+# Force torch CPU ops to use all cores. bf16 matmul on CPU is single-thread on
+# some torch builds even with OMP_NUM_THREADS set; this is the runtime knob that
+# keeps host-side golden compute fast. See test_moe_compute_6U.py for context.
+torch.set_num_threads(os.cpu_count())
```
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_selective_combine_6U.py]: Fixed

File: tests/nightly/tg/ccl/moe/test_moe_compute_utils_tt.py [hunk 2]
```diff
@@ -17,6 +17,11 @@
 import torch
 import ttnn
 
+# Force torch CPU ops to use all cores. bf16 matmul on CPU is single-thread on
+# some torch builds even with OMP_NUM_THREADS set; this is the runtime knob that
+# keeps host-side golden compute fast. See test_moe_compute_6U.py for context.
+torch.set_num_threads(os.cpu_count())
```
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_utils_tt.py]: Fixed

File: tests/nightly/tg/ccl/moe/test_moe_compute_6U.py [hunk 3]
```diff
@@ -115,12 +115,14 @@ def _expand_model_configs(configs):
     MoEModelConfig("deepseek_v3",         N=2048, hidden_size=7168, selected_experts_k=8, has_bias_values=(False, True), test_modes=("perf", "correctness")),
     MoEModelConfig("kimi_k25",            N=2048, hidden_size=7168, selected_experts_k=8, experts_per_device_values=(6,), num_layers=3, num_iterations=2),
     MoEModelConfig("deepseek_v4_flash",   N=2048, hidden_size=4096, selected_experts_k=6),
+
+    # these next few are skipped because they are known failures. Not using xfail for the sake of reducing test runtime.
     MoEModelConfig("deepseek_v4_pro",     N=3072, hidden_size=7168, selected_experts_k=6, experts_per_device_values=(6,), num_layers=3, num_iterations=2,
-                   marks=(pytest.mark.xfail(reason="Combine AllClose fails for specific output values (hidden=7168, N=3072) — likely selective_reduce_combine kernel bug"),)),
+                   marks=(pytest.mark.skip(reason="Combine AllClose fails for specific output values (hidden=7168, N=3072) — likely selective_reduce_combine kernel bug"),)),
```
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_6U.py]: Fixed

File: tests/nightly/tg/ccl/moe/test_moe_compute_6U.py [hunk 4]
```diff
@@ -115,12 +115,14 @@ def _expand_model_configs(configs):
...
     MoEModelConfig("mistral_large_3",     N=4096, hidden_size=7168, selected_experts_k=4, num_layers=3, num_iterations=2,
-                   marks=(pytest.mark.xfail(reason="L1 overflow: N=4096 A2A buffer (12*12*2048=288KB) exceeds Wormhole L1 budget by ~21KB"),)),
+                   marks=(pytest.mark.skip(reason="L1 overflow: N=4096 A2A buffer (12*12*2048=288KB) exceeds Wormhole L1 budget by ~21KB"),)),
```
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_6U.py]: Fixed

File: tests/nightly/tg/ccl/moe/test_moe_compute_6U.py [hunk 5]
```diff
@@ -115,12 +115,14 @@ def _expand_model_configs(configs):
...
     MoEModelConfig("ling_1t",             N=2048, hidden_size=8192, selected_experts_k=8, num_layers=3, num_iterations=2,
-                   marks=(pytest.mark.xfail(reason="Wormhole L1 too small for hidden=8192: dim=4 overflows mux L1 by 93KB, dim=2 overflows combine CB (2MB > 1MB bank)"),)),
+                   marks=(pytest.mark.skip(reason="Wormhole L1 too small for hidden=8192: dim=4 overflows mux L1 by 93KB, dim=2 overflows combine CB (2MB > 1MB bank)"),)),
```
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_6U.py]: Fixed

File: tests/nightly/tg/ccl/moe/test_moe_compute_6U.py [hunk 6]
```diff
@@ -1700,11 +1644,243 @@ def _upload_raw(t):
...
+    # CREATE DEVICE INPUT TENSORS
+    #########################################
+
+    # Drain tilize core: per-arch coordinate where indices/scores are L1-sharded so the
+    # op kernel can read them via NOC. Must match the op's drain tilize core (tilize_cores[0]
+    # in moe_compute_program_factory.cpp's get_layout()), or non-drain tilize cores will
+    # noc_async_read garbage L1 addresses on the drain core (CB overflow caught by watcher).
+    #   WH (max_tilize_cores[0]): (6, 9)
+    #   BH (max_tilize_cores[0]): (10, 9)  — DRAM cols shifted, tilize moved to x=9,10
+    is_blackhole_arch = mesh_device.arch() == ttnn.Arch.BLACKHOLE
```
[the remark this answers is not in the document]
Comment (llongTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_6U.py]: Co-pilot nailed it.

**Review by llongTT — APPROVED** [review]

File: tests/nightly/tg/ccl/moe/test_moe_compute_6U.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (amorrisonTT) [reply tests/nightly/tg/ccl/moe/test_moe_compute_6U.py]: Fixed

**Review by roseli-TT — APPROVED** [review]
